lib/Transforms/LoopFusion.cpp

```diff
@@ -322,6 +322,38 @@ struct MemRefDependenceGraph {
     return false;
   }
 
+  // Returns the unique AffineStoreOp in `node` that meets all the following:
+  //   *) store is the only one that writes to a function-local live out memref,
+  //   *) store is not the source of a self-dependence on `node`.
+  // Otherwise, returns a null AffineStoreOp.
+  AffineStoreOp getUniqueStoreToLiveOut(Node *node) {
+    AffineStoreOp uniqueStore;
+    for (auto *op : node->stores) {
+      auto storeOp = cast<AffineStoreOp>(op);
+      auto *memref = storeOp.getMemRef();
+      auto outEdgeIt = outEdges.find(node->id);
+      // Skip this store if there are no dependences on its memref. This means
+      // that store either:
+      // *) writes to a memref that is only read within the same loop nest
+      //    (self-dependence edges are not represented in graph at the moment),
+      // *) writes to a function live out memref (function parameter), or
+      // *) is dead.
+      if (outEdgeIt == outEdges.end() ||
+          llvm::all_of(outEdgeIt->second, [=](const Edge &edge) {
```

Comment:
Hoist out outEdgeIt->second access as well.
```
const auto &nodeOutEdges = outEdgeIt->second;
```

The only thing needed inside is:
```
   if (llvm::all_of(outEdges, [=](const Edge &edge) { 
       ...
       ) 
       continue;
```